### ISSUE TITLE
Sync Quote Lookup Data to Mobile

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#3411]
**[Mingle Card: 3723](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3723)**
Narrative
---------

As a Field Engineer, I can set lookup values on my Quote Data, so that (see Feature Card).

Acceptance Criteria
-------------------

*   It should download all of the lookups found in the transactional metadata, with some exceptions listed below.
    *   It should not download the same lookups more than once.
        
    *   It should not download lookups of lookups.
        
    *   It should not download transactional data covered in other parts of the sync. This is covered in another card. #3729
        
*   The search result layout needs to be available once the sync is complete. It should only save the search layout when it has changed.
    
*   It should do a partial sync anytime after the initial sync taking deleted, new and changed items into account.
    
*   The lookup data must be searchable using the keyword index. This card is for _fetching _the data, not for storing it (which is a separate card).
    

Analysis
--------

*   We need a way to coordinate the old and new parts of the sync to avoid downloading the same things twice (see card #3729).
    *   We cannot know which lookups to retrieve until the old sync has retrieved the index of lookups in the old sync:
    *   see lookup-query.js line 7, the loadLookups method receives a list of sobject types as a parameter
*   We need to fetch the description before fetching the data so we can use it to build up the query.
*   Fetch of the search result layout during sync should be completely separated from the data.

**Note**: Fetching and saving lookup data can use the same method as the saving transactional data.

Related Stories
---------------

Tasks
-----

{{table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD}}

Defects
-------

{{ table query: SELECT Number, Name, Owner, 'Status' WHERE Type = Defect and 'Related Story' = THIS CARD }}

Test Plan
---------

**Comment by githubbob42** [comment]
### Feature: #3404 CPQ 2 - Syncing ([Mingle](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3716))

**Comment by githubbob42** [comment]
**Pull Request #1612**
